integration-tests/docker/test_env/sol.go

```diff
@@ -0,0 +1,174 @@
+package test_env
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	tc "github.com/testcontainers/testcontainers-go"
+	tcwait "github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/smartcontractkit/chainlink-solana/integration-tests/utils"
+	"github.com/smartcontractkit/chainlink-testing-framework/docker/test_env"
+	"github.com/smartcontractkit/chainlink-testing-framework/logging"
+)
+
+const (
+	SOL_HTTP_PORT = "8899"
+	SOL_WS_PORT   = "8900"
+)
+
+var config_yml = `
+json_rpc_url: http://0.0.0.0:8899
+websocket_url: ws://0.0.0.0:8900
+keypair_path: /root/.config/solana/cli/id.json
+address_labels:
+  "11111111111111111111111111111111": ""
+commitment: finalized
+`
+
+var id_json = `
+[205,246,252,222,193,57,3,13,164,146,52,162,143,135,8,254,37,4,250,48,137,61,49,57,187,210,209,118,108,125,81,235,136,69,202,17,24,209,91,226,206,92,80,45,83,14,222,113,229,190,94,142,188,124,102,122,15,246,40,190,24,247,69,133]
+`
+
+type Solana struct {
+	test_env.EnvComponent
+	ExternalHttpUrl string
+	ExternalWsUrl   string
+	InternalHttpUrl string
+	InternalWsUrl   string
+	t               *testing.T
+	l               zerolog.Logger
+}
+
+func NewSolana(networks []string, opts ...test_env.EnvComponentOption) *Solana {
+	ms := &Solana{
+		EnvComponent: test_env.EnvComponent{
+			ContainerName: fmt.Sprintf("%s-%s", "solana", uuid.NewString()[0:8]),
+			Networks:      networks,
+		},
+		l: log.Logger,
+	}
+	for _, opt := range opts {
+		opt(&ms.EnvComponent)
+	}
+	return ms
+}
+
+func (s *Solana) WithTestLogger(t *testing.T) *Solana {
+	s.l = logging.GetTestLogger(t)
+	s.t = t
+	return s
+}
+
+func (s *Solana) StartContainer() error {
+	l := tc.Logger
+	if s.t != nil {
+		l = logging.CustomT{
+			T: s.t,
+			L: s.l,
+		}
+	}
+	cReq, err := s.getContainerRequest()
+	if err != nil {
+		return err
+	}
+	c, err := tc.GenericContainer(context.Background(), tc.GenericContainerRequest{
+		ContainerRequest: *cReq,
+		Reuse:            true,
+		Started:          true,
+		Logger:           l,
+	})
+	if err != nil {
+		return errors.Wrapf(err, "cannot start Solana container")
```

Comment:
Would std lib errors be sufficient here?
```suggestion
		return fmt.Errorf("cannot start Solana container: %w", err)
```